scripts/morty/analysis.py

```diff
@@ -0,0 +1,273 @@
+import solvers
+import numpy as np
+from data import DataHandler
+from scipy import integrate
+
+
+class AnalysisCollection:
+    def __init__(self, analysis_params, run_params, data_params):
+        """
+        This class contains a collection of analysis procedures which
+        can be run.
+        The data is returned in an easily-plottable dictionary as
+        well as saved to
+        text files for future analysis.
+
+        Parameters
+        ----------
+        analysis_params : dict
+            key : str
+                Name of analysis parameter
+        run_params : dict
+            key : str
+                Name of run parameter
+        data_params : dict
+            ket : str
+                Name of data parameter
+        """
+        self.analysis_params = analysis_params
+        self.run_params = run_params
+        self.data_params = data_params
+        return
```

Comment:
Throughout your code, you have unadorned `return`s at the ends of functions. I recommend either not including return at all (Simple is better than complex.) or being explicit `return None` (Explicit is better than implicit.)



https://stackoverflow.com/a/15300671